src/components/sidebar/types.d.ts

```diff
@@ -0,0 +1,15 @@
+type NodeT = import("react").ReactNode
+
+interface SidebarProps {
+  info?: NodeT
+  right?: boolean
+  children?: NodeT
+}
+
+interface PortalSidebarProps<T = any> {
+  isOpen?: boolean
+  closeOnEsc?: boolean
+  onClose?: (args?: T) => void
```

Comment:
Wouldn't it be better to have a `onToggle` with a `isOpen` boolean param so we have a handler also when it opens?

Reply:
@VLegakis  `isOpen` is state, `onToggle` as far it is sidebar all kind of controls for it is in the parent component

Reply:
isOpen removed. As far as all controls will be in the parent component.

Reply:
So, if `isOpen` is removed and the controls reside on the parent, what does `onClose` do? Can you elaborate a bit on this?

Reply:
side effect wich will be fired on close